File: files/CSEC 08.09.py
import pandas as pd

path = "files/Excel Challenge September 8th.xlsx"
input = pd.read_excel(path, usecols = "B:C", skiprows = 1, nrows = 5, names = ["Stall","Stock"])

search_vec = ["Apple", "Kiwi"]
result = input
result["Stock"] = result["Stock"].str.replace(" ", "").str.split(",")
result = result.explode("Stock")
result = result[result["Stock"].isin(search_vec)]["Stall"].nunique()

print(result) # 3 
# True

File: files/2024-09-08/CSEC 08.09.R
library(tidyverse)
library(readxl)

path = "files/Excel Challenge September 8th.xlsx"
input = read_excel(path, range = "B2:C7")

search_vec = list("Apple", "Kiwi")

result = input %>%
  mutate(Stock = str_remove_all(Stock, "\\s") %>% str_split(",")) %>%
  mutate(Found = map_lgl(Stock, ~any(search_vec %in% .x))) %>%
  summarise(Count = sum(Found))

result
#   3
#   
#   True